### PR TITLE
Move filter fields into the `FilterMode` structure

### PR DESCRIPTION
**Pull request by fadeevab**
1. Dissect filtering logic to the `FilterMode` implementation.
2. Move filtering to the `on()` event handling.
4. Make `RadioButton` and `Checkbox` private (not sure why they were `pub`).

Continuation of feature #6

### CODE REVIEW

**Comment by fadeevab** [comment]
@cylewitruk  @antoinePutmans Check it out. The diff is not pretty, it's better to view the resulting file.

**Comment by antoinePutmans** [comment]
Hey @fadeevab, it looks way better ! I started implementing it but it seems that you are faster than me haha 😄
Good job ! 👍 

I was wondering if it would be possible to use this FilterMode in the multiselect also 🤔 The `Checkbox<T>` is a `Radiobutton<T>` with a `selected` field added

**Comment by fadeevab** [comment]
Thanks, folks! 

@cylewitruk There is one improvement that it shows an error now if the list appears to be empty after pressing `Enter`:
image [<-Enter]

@antoinePutmans Yeah, I was wondering too, it should be possible but much harder 😅

**Comment by fadeevab** [comment]
@antoinePutmans Actually, maybe not much harder, maybe I can do it 🤔

**Comment by antoinePutmans** [comment]
@fadeevab Be careful there is a problem when pressing the top/bottom keys when the filter result is empty ! 

https://github.com/fadeevab/cliclack/assets/113629846/25fe4cea-d18c-499e-b93f-c6360b4ba9ea

File: src/select.rs [hunk 1]
```diff
@@ -85,51 +87,37 @@ where
...
+            Key::ArrowUp | Key::ArrowLeft => {
+                if *cursor > 0 {
+                    *cursor -= 1;
                 }
             }
-            Key::ArrowLeft => {
-                if self.input_filter.is_empty() && self.cursor > 0 {
-                    self.cursor -= 1;
+            Key::ArrowDown | Key::ArrowRight => {
+                if *cursor < items.len() - 1 {
```
Comment (antoinePutmans) [inline src/select.rs:115]: Should check if items.len() != 0 otherwise it will panic

Comment (fadeevab) [reply src/select.rs:115]: oh, thanks, I'll check it out today later

Comment (fadeevab) [reply src/select.rs]: Fixed

**Comment by fadeevab** [comment]
Superseded by PR #52

**Pull request closed without merging**